src/main/java/edu/usfca/numberguessgame/service/GameService.java

```diff
@@ -1,27 +1,33 @@
 package edu.usfca.numberguessgame.service;
 
-import edu.usfca.numberguessgame.model.GuessRequest;
 import edu.usfca.numberguessgame.model.Session;
+import edu.usfca.numberguessgame.repository.SessionRepository;
+
+import java.util.Optional;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.mongodb.core.MongoTemplate;
 import org.springframework.stereotype.Service;
-import org.springframework.ui.Model;
 
 @Service
 public class GameService {
 
     @Autowired
-    private MongoTemplate mongoTemplate;
+    private SessionRepository sessionRepository;
 
     /**
      * return -1 if the input is not an integer it is smaller than 0
      * return number if the input is valid
      */
     public int validateUserInput(String input) {
-        if (input.matches("[0-9]+") && Integer.parseInt(input) >= 0){
-            return Integer.parseInt(input);
-        }else{
+        int parseInt;
```

Comment:
Still awkward code.
The first attempt was cleaner but still redundant.
Also remember to use the inverse for cleaner if statements.
```
public int validateUserInput(String input) {
    int parsed;
    if (!input.matches("[0-9]+") || (parsed = Integer.parseInt(input)) < 0){
        throw new RuntimeException("some message") // or can be return 1;
    }
    return parsed;
}
```